app/models/match.rb

```diff
@@ -0,0 +1,7 @@
+class Match < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :matcher, class_name: 'User'
```

Comment:
I might be tempted to user matcher and matchee (or something) here rather than leaving one as the default "user."